### ISSUE TITLE
tpm2_ptool add link option

### ISSUE THREAD

**Issue by Koerk123** [tpm2-software/tpm2-pkcs11#485]
I created a rsa key using the following command in a Makefile:
`tpm2tss-genkey -a rsa -s $(GENRSA_NUMBITS) $(PRIVATE)/$@` 
The created file looks like:
```
-----BEGIN TSS2 PRIVATE KEY-----
Something
-----END TSS2 PRIVATE KEY-----
```
Then I tried to import the key using tpm2_ptool into the token I mentioned in #480 
This led to an error:
```
sudo ./tpm2_ptool import --userpin **** --privkey /home/pi/TPM/TPM2_Certs/root/ca/private/root.key.tss --label **** --algorithm rsa --path /home/pi/tmp/
command: tpm2_import -V -C /tmp/tmpzguoyx39/primary.handle -i /home/pi/TPM/TPM2_Certs/root/ca/private/root.key.tss -u /tmp/tmpzguoyx39/eleoxpye.pub -r /tmp/tmpzguoyx39/wfgwm6ts.priv -P d07b4525d96abeb58ea66aa867e24479 -p 562294eff574648d1b2e615c32566a92 -G rsa
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/pi/TPM/tpm2-pkcs11/tools/tpm2_pkcs11/tpm2_ptool.py", line 30, in <module>
    main()
  File "/home/pi/TPM/tpm2-pkcs11/tools/tpm2_pkcs11/tpm2_ptool.py", line 26, in main
    commandlet.init('A tool for manipulating the tpm2-pkcs11 database')
  File "/home/pi/TPM/tpm2-pkcs11/tools/tpm2_pkcs11/command.py", line 102, in init
    commandlet.get()[d['which']](d)
  File "/home/pi/TPM/tpm2-pkcs11/tools/tpm2_pkcs11/commandlets_keys.py", line 230, in __call__
    objects = super(ImportCommand, self).__call__(args)
  File "/home/pi/TPM/tpm2-pkcs11/tools/tpm2_pkcs11/commandlets_keys.py", line 174, in __call__
    pobj, objauth, tpm2, path, alg, privkey, d)
  File "/home/pi/TPM/tpm2-pkcs11/tools/tpm2_pkcs11/commandlets_keys.py", line 225, in new_key_create
    tr_handle, pobj['objauth'], objauth, privkey=privkey, alg=alg)
  File "/home/pi/TPM/tpm2-pkcs11/tools/tpm2_pkcs11/tpm2.py", line 236, in importkey
    str(stderr))
RuntimeError: Could not execute tpm2_import: b'INFO on line: "373" in file: "lib/files.c": Assuming tpm context file\n1996419088:error:0909006C:PEM routines:get_name:no start line:../crypto/pem/pem_lib.c:745:Expecting: ANY PRIVATE KEY\nERROR on line: "1033" in file: "lib/tpm2_openssl.c": Reading PEM file "/home/pi/TPM/TPM2_Certs/root/ca/private/root.key.tss" failed\nERROR on line: "146" in file: "tools/tpm2_tool.c": Unable to run tpm2_import\n'
```
The error also occurs when I use `--id` / `--key-label`.
Any idea what might cause the error?
Thank you.

**Comment by williamcroberts** [comment]
That doesn't work currently, you need to use an actual key where you have the key bytes, ie something like `openssl genrsa`

**Comment by williamcroberts** [comment]
really what we want here is a tpm2_ptool link function that can take a:
1. tpm2tss-genkey key
2. tpm2_create generated key

and load it under the parent key, (pid) provided by the token label. It would have to invoke a load to make sure that it already belongs to that primary object.

This would allow folks with existing hierarchies and objects to get access to them.

**Comment by Koerk123** [comment]
Thank you for replying.
That means I can create keys using `openssl genrsa` and be able to import them.
Is that as well possible when I pass `-engine` parameter?
Certificates created with the `openssl genrsa` key are also possible to be imported then?

**Comment by williamcroberts** [comment]
Im re-opening this, becuase it has the link functionality documented and we need to do it.

**Comment by williamcroberts** [comment]
I see a few things, here the default for the tss engine tool is to create a transient ECC parent, so we would need support for tokens with transient EKs. Then we need to be able to parse the ASN1 structure and pull in the parent handle (if set) and the object itself.

Ill start with adding support for existing primary objects.

**Comment by williamcroberts** [comment]
Fixed on master via PRs: #539, #543, #545
